sdk/nodejs/src/utils/userError.ts

```diff
@@ -0,0 +1,7 @@
+export class UserError extends Error {
+    constructor(error: Error | string) {
+        super(error instanceof Error ? error.message : error)
+        this.name = 'UserError'
+        this.stack = error instanceof Error ? error.stack : undefined
```

Comment:
does this make it so the stack isnt set properly when it's thrown with a string message?

Reply:
this is for the cases where we throw an error when we know something is wrong and we throw with a string message